templates/app_client/lib/app_client/api_version/adapter.rb.erb

```diff
@@ -18,16 +18,18 @@ module <%= gem_name_pascal %>
           access_token  ||= config.access_token
           api_key       ||= config.api_key
           api_secret    ||= config.api_secret
-          options         = config.options.merge!(options)
+          url             = "#{config.base_url}/<%= resource_url %>"
 
           # Workaround issue with some TLS setups (haproxy)
           # Prevents :443 being added to hostname for TLS
           Excon.defaults[:omit_default_port] = true
 
+          options = { url: url }.merge(options)
+
           if access_token
-            config.adapter_factory.connect_oauth(access_token, options)
+            Adapters::Http.connect_oauth(access_token, options)
           elsif api_key && api_secret
-            config.adapter_factory.connect(api_secret, options.merge!({
+            Adapters::Http.connect(api_secret, options.merge!({
```

Comment:
I think this will give you the root-level Heroics client, rather than the one for the specific resource, so I think the `adapter_factory` thing probably needs to be preserved, otherwise this is probably a breaking change.

Reply:
I think this will resolve to the full, versioned adapter: `ChirpClient::V4::Adapters::HTTP` right?

Reply:
Right, but that's the Heroics client for the whole V4 Chirp API. Not just one resource. These adapter classes are per-resource

Reply:
So this might give you a client that has the method `patients` and that's it. I.e. you wouldn't be able to do `show`, `list`, etc. You'd have to do `patients.list`, which would break all other gutsy-generated clients, which use the per-resource clients from heroics.

Reply:
This should always resolve to the same class as the `adapter_factory` method in `master`:
https://github.com/IoraHealth/gutsy/blob/master/templates/app_client/lib/app_client.rb.erb#L58

Right? It's just using module nesting instead of a `const_get`

Reply:
Oh shittt... yeah, you're right. Sorry, I was forgetting how this all works. Forgot about how the `method_missing` does the delegation to the resource clients. Looks good to me :)

Reply:
Com'on MAX

Reply:
I AM BLAMING JET LAG